internal/p2p/router.go

```diff
@@ -417,7 +417,7 @@ func (r *Router) routeChannel(
 
 func (r *Router) numConcurrentDials() int {
 	if r.options.NumConcurrentDials == nil {
-		return runtime.NumCPU()
+		return runtime.NumCPU() * 32
```

Comment:
Could we have this as a constant somewhere i.e. `const defaultConcurrentDialsFactor = 32`